tests/api/cyto_dl_model/test_segmentation_plugin_model.py

```diff
@@ -0,0 +1,123 @@
+# are actually used how we expect by Cyto-DL, just that they exist in the default config.
+@pytest.fixture
+def model_with_default_config() -> SegmentationPluginModel:
+    return SegmentationPluginModel.from_default_config(3)
+
+
+@pytest.fixture
+def model_with_bad_config() -> SegmentationPluginModel:
+    return SegmentationPluginModel.from_existing_config(Path(__file__).parent / "bad_config.yaml")
+
+
+class TestDefaultConfig:
+    @patch("cyto_dl.api.cyto_dl_model.cyto_dl_base_model.train_model")
+    def test_train_no_ckpt(
+        self, train_model_mock: Mock, model_with_default_config: SegmentationPluginModel
+    ):
+        model_with_default_config.train(1, "manifest", "output_dir")
+        train_model_mock.assert_called_once()
+
+    @patch("cyto_dl.api.cyto_dl_model.cyto_dl_base_model.train_model")
+    def test_train_with_ckpt(
+        self, train_model_mock: Mock, model_with_default_config: SegmentationPluginModel
+    ):
+        model_with_default_config.train(1, "manifest", "output_dir", checkpoint=Path("ckpt"))
+        train_model_mock.assert_called_once()
+
+    @patch("cyto_dl.api.cyto_dl_model.cyto_dl_base_model.evaluate_model")
+    def test_predict(
+        self, evaluate_model_mock: Mock, model_with_default_config: SegmentationPluginModel
+    ):
+        model_with_default_config.predict("manifest", "output_dir", Path("ckpt"))
+        evaluate_model_mock.assert_called_once()
+
+    def test_input_channel(self, model_with_default_config: SegmentationPluginModel):
+        assert model_with_default_config.get_input_channel() is not None
+        model_with_default_config.set_input_channel(785)
+        assert model_with_default_config.get_input_channel() == 785
+
+    def test_raw_image_channels(self, model_with_default_config: SegmentationPluginModel):
+        assert model_with_default_config.get_raw_image_channels() is not None
+        model_with_default_config.set_raw_image_channels(35)
+        assert model_with_default_config.get_raw_image_channels() == 35
+
+    def test_manifest_column_names(self, model_with_default_config: SegmentationPluginModel):
+        assert len(model_with_default_config.get_manifest_column_names()) == 6
+        model_with_default_config.set_manifest_column_names("a", "b", "c", "d", "e", "f")
+        assert model_with_default_config.get_manifest_column_names() == (
+            "a",
+            "b",
+            "c",
+            "d",
+            "e",
+            "f",
+        )
+
+    def test_split_column(self, model_with_default_config: SegmentationPluginModel):
+        assert model_with_default_config.get_split_column() is None
+        model_with_default_config.set_split_column("foo")
+        assert model_with_default_config.get_split_column() == "foo"
+        model_with_default_config.remove_split_column()
+        assert model_with_default_config.get_split_column() is None
+
+
+class TestBadConfig:
+    @patch("cyto_dl.api.cyto_dl_model.cyto_dl_base_model.train_model")
```

Comment:
In all of these tests I am unclear about what is bad about the model.  Can you comment on that?

Reply:
In these tests, the config we load is what is 'bad'. The 'bad' config I created contains none of the expected key-paths (e.g. `trainer.max_epochs` does not exist in this config).

I think of `TestBadConfig` as a sanity check: if the `TestBadConfig` tests fail, then we can't be confident that the `TestDefaultConfig` tests will fail when they should (when a change occurs to the config structure and the API hasn't been modified accordingly)